src/mesh/p4est_mesh.jl

```diff
@@ -139,22 +226,61 @@ function connectivity_structured_periodic(cells_x, cells_y)
   vertices = C_NULL
   tree_to_vertex = C_NULL
 
-  # Periodic boundaries
   tree_to_tree = Matrix{p4est_topidx_t}(undef, 4, num_trees)
+  tree_to_face = Matrix{Int8}(undef, 4, num_trees)
+
   for cell_y in 1:cells_y, cell_x in 1:cells_x
     tree = linear_indices[cell_x, cell_y]
+
     # Subtract 1 because p4est uses zero-based indexing
-    tree_to_tree[1, tree] = linear_indices[mod(cell_x - 2, cells_x) + 1, cell_y] - 1
-    tree_to_tree[2, tree] = linear_indices[mod(cell_x, cells_x) + 1, cell_y] - 1
-    tree_to_tree[3, tree] = linear_indices[cell_x, mod(cell_y - 2, cells_y) + 1] - 1
-    tree_to_tree[4, tree] = linear_indices[cell_x, mod(cell_y, cells_y) + 1] - 1
-  end
+    # Negative x-direction
+    if cell_x > 1
+      tree_to_tree[1, tree] = linear_indices[cell_x - 1, cell_y] - 1
+      tree_to_face[1, tree] = 1
+    elseif periodicity[1]
+      tree_to_tree[1, tree] = linear_indices[cells_x, cell_y] - 1
+      tree_to_face[1, tree] = 1
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[1, tree] = tree - 1
+      tree_to_face[1, tree] = 0
+    end
 
-  tree_to_face = Matrix{Int8}(undef, 4, num_trees)
-  tree_to_face[1, :] .= 1
-  tree_to_face[2, :] .= 0
-  tree_to_face[3, :] .= 3
-  tree_to_face[4, :] .= 2
+    # Positive x-direction
+    if cell_x < cells_x
+      tree_to_tree[2, tree] = linear_indices[cell_x + 1, cell_y] - 1
+      tree_to_face[2, tree] = 0
+    elseif periodicity[1]
+      tree_to_tree[2, tree] = linear_indices[1, cell_y] - 1
+      tree_to_face[2, tree] = 0
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[2, tree] = tree - 1
+      tree_to_face[2, tree] = 1
+    end
+
+    # Negative x-direction
+    if cell_y > 1
+      tree_to_tree[3, tree] = linear_indices[cell_x, cell_y - 1] - 1
+      tree_to_face[3, tree] = 3
+    elseif periodicity[2]
+      tree_to_tree[3, tree] = linear_indices[cell_x, cells_y] - 1
+      tree_to_face[3, tree] = 3
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[3, tree] = tree - 1
+      tree_to_face[3, tree] = 2
+    end
+
+    # Positive x-direction
+    if cell_y < cells_y
+      tree_to_tree[4, tree] = linear_indices[cell_x, cell_y + 1] - 1
+      tree_to_face[4, tree] = 2
+    elseif periodicity[2]
+      tree_to_tree[4, tree] = linear_indices[cell_x, 1] - 1
+      tree_to_face[4, tree] = 2
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[4, tree] = tree - 1
+      tree_to_face[4, tree] = 3
+    end
+  end
 
   tree_to_corner = C_NULL
   ctt_offset = Array{p4est_topidx_t}([0])
```

Comment:
Just seeing this now - what exactly is this supposed to be?

Reply:
From the p4est docs: `Corner-to-tree offsets (num_corners + 1 values). This must always be non-NULL; in trivial cases it is just a pointer to a p4est_topix value of 0.`
This is a trivial case (I guess because we don't need corner connectivity), so `0` works fine.

Reply:
OK. Maybe just add a comment for future me then :-)